main/src/library/DelayMap.flix

```diff
@@ -378,13 +378,60 @@ namespace DelayMap {
     /// Helper function for `mapWithKey`. Applies `f` eagerly.
     ///
     def mapWithKeyE(f: (k, v1) -> v2 & ef, m: DelayMap[k, v1]): DelayMap[k, v2] & ef =
-        let DMap(t) = m;
         let g = (k, v) -> {
                 let x1 = f(k, force v);
                 lazy x1
             };
+        let DMap(t) = parallelForce(m);
         DMap(RedBlackTree.mapWithKey(g, t))
 
+    ///
+    /// Forces `m` in parallel if it is big, otherwise returns `m`.
+    ///
+    @Parallel
+    def parallelForce(m: DelayMap[k, v]): DelayMap[k, v] =
+        if (useParallelEvaluation(m))
+            forceAll(m)
+        else
+            m
+
+    ///
+    /// Forces **all values** in `m`.
+    ///
+    @Parallel
+    pub def forceAll(m: DelayMap[k, v]): DelayMap[k, v] =
```

Comment:
I was wondering.. Maybe forceAll should return unit? After all its not a new tree. This would make its API more clear, imo.

Reply:
Agreed